### PR TITLE
Name syntax

### PR DESCRIPTION
**Pull request by laurayuwen**
improve name-syntax.k, this part had problems strenching the parser in the past, but somehow it is better now, I am allowed to write all syntax acording to sdf but only cannot use the 'MethodName klabel attribute. And this is probably not related to parser, because it kompiles and even the static semantics creates the correct output. I might not be able to solve this with current version of K, where computation in k cell stucks is not giving me good hint. 
I also found that labels-exec.k and labels-prep.k are not needed now. This is pretty much everything for pure syntax.

### CODE REVIEW

File: src/common/name-syntax.k [hunk 1]
```diff
@@ -1,17 +1,22 @@
...
+syntax AmbName ::= Id                                   [klabel('AmbName)]
+                 | AmbName "." Id                       [klabel('AmbName)]
+
+syntax PackageOrTypeName ::= Id                         [klabel('PackageOrTypeName)]
+                           | PackageOrTypeName "." Id   [strict(1), klabel('PackageOrTypeName)]
+
+syntax ExprName::= Id                                   [klabel('ExprName)]
+                 | AmbName "." Id                       [klabel('ExprName)]
+
+syntax TypeName ::= Id                                  [klabel('TypeName)]
```
Comment (andreistefanescu) [inline src/common/name-syntax.k]: Is this the same with ```PackageOrTypeName```?

File: src/common/name-syntax.k [hunk 2]
```diff
@@ -1,17 +1,22 @@
...
+
+syntax PackageName ::= IdList                           [klabel('PackageName)]
+
+syntax AmbName ::= Id                                   [klabel('AmbName)]
+                 | AmbName "." Id                       [klabel('AmbName)]
+
+syntax PackageOrTypeName ::= Id                         [klabel('PackageOrTypeName)]
+                           | PackageOrTypeName "." Id   [strict(1), klabel('PackageOrTypeName)]
+
+syntax ExprName::= Id                                   [klabel('ExprName)]
```
Comment (andreistefanescu) [inline src/common/name-syntax.k]: Is this the same with ```AmbName```?

File: src/common/name-syntax.k (hunk 1 above)
Comment (laurayuwen) [inline src/common/name-syntax.k]: Yes, the same;

context-free syntax
    Id             -> AmbName {cons("AmbName")}
    AmbName "." Id -> AmbName {cons("AmbName")}

    Id                       -> TypeName {cons("TypeName")}
    PackageOrTypeName "." Id -> TypeName {cons("TypeName")}

    Id             -> ExprName {cons("ExprName")}
    AmbName "." Id -> ExprName {cons("ExprName")}

    Id             -> MethodName  {cons("MethodName")}
    AmbName "." Id -> MethodName  {cons("MethodName")}

    Id                       -> PackageOrTypeName {cons("PackageOrTypeName")}
    PackageOrTypeName "." Id -> PackageOrTypeName {cons("PackageOrTypeName")}

**Comment by andreistefanescu** [comment]
lgtm

**Pull request merged**